corehq/apps/case_search/tests/utils.py

```diff
@@ -0,0 +1,9 @@
+from corehq.apps.case_search.models import criteria_dict_to_criteria_list
+from corehq.apps.case_search.utils import CaseSearchQueryBuilder
+
+
+def get_case_search_query(domain, case_types, criteria_dict):
+    """Helper function for tests. In the future this will handle conversion
+    of the criteria from a dict to a list of Criteria objects"""
+    criteria = criteria_dict_to_criteria_list(criteria_dict)
+    return CaseSearchQueryBuilder(domain, case_types).build_query(criteria)
```

Comment:
It looks like `criteria_dict_to_criteria_list` is only used to transform the arg passed to `build_query`, should it be moved inside `build_query` or do you prefer having that be the thing that's passed around/instantiated?

Reply:
The main impetus for this refactor is so that we can pass more meta data to the builder for each key e.g. `multivalue_mode`. This extra data is going to come in as normal POST params via the API so I want to process it before giving it to the builder.